mustache_test.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+    . "github.com/onsi/ginkgo"
+    . "github.com/onsi/gomega"
+    "github.com/onsi/gomega/gexec"
+    "github.com/onsi/gomega/gbytes"
+    "os/exec"
+)
+
+var _ = Describe("Mustache", func() {
+    successDecodeOutput := map[string]interface{}{"key":"val"}
+    expectedTemplate := "my name is Montoooorb69 but you can call me Monica Rachel Pheobe.\n" +
+        "I come from Cloooooooooooooooo in the Florbatorb galaxy.\n" +
+        "I would like to Drink earth liquids, Kill the producers of 3rd Rock From The Sun, Go to Disney World.\n\n" +
+        "Thank you,\nMontoooorb69"
+
+    assertExampleSuccess := func(path string) {
+        command := exec.Command(path)
+        session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+        Expect(err).ShouldNot(HaveOccurred())
+        Eventually(session.Out).Should(gbytes.Say(expectedTemplate))
+        Eventually(session).Should(gexec.Exit(0))
+    }
+
+    Describe("Cli", func() {
+        Context("With data file", func() {
+            It("should successfully output example template from JSON data source", func() {
+                assertExampleSuccess("./examples/run-example-json")
+            })
+
+            It("should successfully output example template from YAML data source", func() {
+                assertExampleSuccess("./examples/run-example-yaml")
```

Comment:
May want to put a note at the top of these examples if your tests depend on them.  I'm not sure people would expect that.  I would expect there to be a test script built inside this file or at least add some scripts to the test directory